doas.c

```diff
@@ -174,11 +175,36 @@ parseconfig(const char *filename, int checkperms)
 	}
 
 	yyparse();
+	yyfn = NULL;
+
 	fclose(yyfp);
 	if (parse_errors)
 		exit(1);
 }
 
+#ifdef DOAS_CONFDIR
+static void
+parseconfdir(const char *dirpath, int checkperms)
+{
+	DIR *dir;
+	struct dirent *dirent;
+	char pathbuf[PATH_MAX];
+
+	dir = opendir(dirpath);
+	if (!dir)
+		err(1, checkperms ? "doas is not enabled, %s" :
+		    "could not open config directory %s", dirpath);
+
+	for (dirent = readdir(dir); dirent; dirent = readdir(dir))
+	{
```

Comment:
`opendir(3)`/`readdir(3)` is not sorted right? This should be sorted alphanumerically because the order of rules is important.

Reply:
Ordering of `opendir`/`readdir` is filesystem-defined, yes.

Reply:
Ok, it has always been there: https://man.bsd.lv/scandir

Reply:
Yes, I was thinking to use scandir.